### PR TITLE
feat: Implement audio groups

### PR DESCRIPTION
**Pull request by zugz**
This change is [Reviewable](https://reviewable.io/reviews/toktok/toxic/59)

### CODE REVIEW

File: src/audio_call.c [hunk 1]
```diff
@@ -61,6 +61,8 @@
 extern FriendsList Friends;
 extern ToxWindow *windows[MAX_WINDOWS_NUM];
 
+extern pthread_mutex_t tox_lock;
```
Comment (iphydf) [inline src/audio_call.c:64]: Can you document what specifically is guarded by this lock?

File: src/chat_commands.c [hunk 2]
```diff
@@ -126,17 +126,30 @@ void cmd_join_group(WINDOW *window, ToxWindow *self, Tox *m, int argc, char (*ar
         return;
     }
 
-    if (type != TOX_CONFERENCE_TYPE_TEXT) {
-        line_info_add(self, NULL, NULL, NULL, SYS_MSG, 0, 0, "Toxic does not support audio groups.");
-        return;
-    }
+    uint32_t groupnum ;
```
Comment (iphydf) [inline src/chat_commands.c:129]: nit: no space before ;

File: src/execute.c [hunk 3]
```diff
@@ -98,20 +98,21 @@ static struct cmd_func chat_commands[] = {
...
 
 static struct cmd_func group_commands[] = {
-    { "/title",     cmd_set_title   },
+    { "/title",     cmd_set_title     },
 
 #ifdef AUDIO
-    { "/mute",      cmd_mute        },
-    { "/sense",     cmd_sense       },
+    { "/audio",     cmd_enable_audio  },
+    { "/mute",      cmd_group_mute },
```
Comment (iphydf) [inline src/execute.c:105]: nit: align braces

**Comment by JFreegman** [comment]
Not quite ready to merge yet. I caught a SIGSEGV when I closed an audio group the first time I tried to test it. Will investigate further when I have time to do more thorough testing.

**Comment by JFreegman** [comment]
Current issues:
- The segfault isn't fixed (https://gist.github.com/JFreegman/d1ec387ea9c17da57f0ec31e3985813f)
- `/mute` with no argument doesn't always work
- `/mute` should print a message indicating that so and so is muted or unmuted
- Tab completion should work for `/mute`
- The group section of the help menu should have entries for `/lsdev` and `/sdev` (audio commands should be split up the same way they are in the chat help menu)
- What happens if two peers share the same nick?

File: src/groupchat.c [hunk 4]
```diff
@@ -756,3 +1044,145 @@ static ToxWindow *new_group_chat(uint32_t groupnum)
...
+bool group_mute_peer(const Tox *m, uint32_t groupnumber, uint32_t peernum)
+{
+    if (tox_conference_peer_number_is_ours(m, groupnumber, peernum, NULL)) {
+        return group_mute_self(groupnumber);
+    }
+
+    GroupChat *chat = &groupchats[groupnumber];
+
+    if (!chat->active || !chat->audio_enabled
+            || peernum < 0 || peernum > chat->max_idx) {
```
Comment (iphydf) [inline src/groupchat.c:1175]: peernum is uint32_t, it can't be < 0.

**Comment by robinlinden** [comment]
Merged in #63

**Pull request closed without merging**